flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adapter/DefaultExecutionTopology.java

```diff
@@ -91,6 +88,8 @@ public DefaultExecutionTopology(ExecutionGraph graph) {
 		this.pipelinedRegionsByVertex = new HashMap<>();
 		this.pipelinedRegions = new ArrayList<>();
 		initializePipelinedRegions();
+
+		ensureCoLocatedVerticesInSameRegion();
```

Comment:
Maybe a side note not for this PR: I would put `initializePipelinedRegions` and `ensureCoLocatedVerticesInSameRegion` into a factory method `fromExecutionGraph`.

Reply:
Yes I agree that publishing `this` in the constructor is not a good idea.
Will add a separate commit to fix it.

Reply:
done.